server/protocols/jmap/src/main/java/org/apache/james/jmap/model/UpdateMessagePatch.java

```diff
@@ -122,12 +146,40 @@ public UpdateMessagePatch build() {
         return isAnswered;
     }
 
-    public boolean isFlagsIdentity() {
+    public Optional<Set<JmapFlag>> getKeywords() {
+        return keywords;
+    }
+
+    public boolean isFlagsIdentity(Flags currentFlags) {
```

Comment:
As explained before, you are not supposed to depend on the input here to determine if you are identity or not.

You should concider the *keyword optional being empty and the is\* properties being empty* as your identity condition.